modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/EditJournalArticleContentDashboardItemAction.java

```diff
@@ -66,10 +74,20 @@ public Type getType() {
 
 	@Override
 	public String getURL() {
+		Portlet portlet = _portletLocalService.getPortletById(
+			_portal.getPortletId(_httpServletRequest));
+
+		HttpServletRequest httpServletRequest =
```

Comment:
Hi @cgoncas! why do we need to create a new request here? in which situation should I use this method? thx

Reply:
Hey @darquesdev !!

As a rule of thumb, you should never use it :)

![image](https://user-images.githubusercontent.com/5776822/99383754-a0bc4880-28ce-11eb-982d-242e5c37a044.png)

If you are curious about why do we need to do it here, please, take a look at how the backlink is generated in the Web Content Editor.

I think this is a really good exercise of how the portal sometimes works, so please, add it to your Liferay learning plan. We can talk about it in our 1:1.

Thanks!!!

Reply:
hahaha, Ok, added to my learning plan, thanks!